bottom-navigator/src/main/java/com/pandora/bottomnavigator/FragmentTransactionHandler.kt

```diff
@@ -236,19 +245,63 @@ internal data class TagStructure private constructor(
         private const val OURTAG = "com.pandora.navigator"
         private const val SEPARATOR = "|"
         private const val DETACHABLE = "DETACHABLE"
+        private const val TRANSITIONS = "TRANSITIONS"
 
         fun fromTag(tag: String?): TagStructure {
             if (tag == null || !tag.startsWith(OURTAG)) {
-                return TagStructure(null, null, null)
+                return TagStructure(null, null, null, null)
                     .apply { isOurFragment = false }
             }
 
-            val (ourTag, className, detachable, uuid) = tag.split(SEPARATOR)
+            val (tagData, transitions) = TransitionsData.fromTag(tag)
+            val (ourTag, className, detachable, uuid) = tagData.split(SEPARATOR)
 
-            if (ourTag != OURTAG) return TagStructure(null, null, null)
+            if (ourTag != OURTAG) return TagStructure(null, null, null, null)
                 .apply { isOurFragment = false }
 
-            return TagStructure(className, DETACHABLE == detachable, uuid)
+            return TagStructure(className, DETACHABLE == detachable, uuid, transitions)
+        }
+    }
+
+    /**
+     * Data class for a fragment's transitions. Responsible for serializing and de-serializing itself
+     */
+    internal data class TransitionsData(
+        val enterAnim: Int,
+        val exitAnim: Int,
+        val popEnterAnim: Int,
+        val popExitAnim: Int
+    ) {
+        override fun toString(): String {
+            return StringBuilder()
+                .append(TRANSITIONS, SEPARATOR
+                    , enterAnim, SEPARATOR
+                    , exitAnim, SEPARATOR
+                    , popEnterAnim, SEPARATOR
+                    , popExitAnim
+                ).toString()
+        }
+
+        companion object {
+            /**
+             * Extracts the transitions data from the tag and returns a pair with transitions and
+             * the rest of the tag
+             */
+            fun fromTag(tag: String): Pair<String, TransitionsData?> {
+                return if (!tag.contains(TRANSITIONS)) {
+                    Pair(tag, null)
+                } else {
+                    val (remainingTag, transitions) = tag.split(TRANSITIONS)
```

Comment:
I don't like this scheme because it relies on TransitionsData structure being at the end of the tag. The next developer that comes along trying to add a field might not realize that transitions has to stay at the end. 

Also you don't need the string "TRANSITIONS" in the tag since the whole tag is serialized as kind of a positional array, meaning that we know what each field is for based on its position in the list. So one option is you could put the 4 new ints as 4 new fields in the string

```com.pandora.navigator|<classname string>|DETACHABLE|<uuid string>|<enterAnim>|<exitAnim>|<popEnterAnim>|<popExitAnim>```

But the problem with that is that the whole TransitionData can be null so maybe it's better to keep it as a single field in the overall tag and pick a new separator for inside that field. For example if you use a period as the TransitionData separator

```com.pandora.navigator|<classname string>|DETACHABLE|<uuid string>|<enterAnim>.<exitAnim>.<popEnterAnim>.<popExitAnim>```

Then you can keep splitting the whole tag simple as before, and then split the TransitionData after you get it from the full tag

```kotlin 
val (ourTag, className, detachable, uuid, transitionData) = tagData.split(SEPARATOR)

val (openEnterAnim, openExitAnim, closeEnterAnim, closeExitAnim) = transitionData.split(".")
```

Reply:
Good points, this simplifies deserializing the transitions too. I don't like that the next dev will have to find a new unique separator, but using a different separator is still cleaner and more sustainable than adding the TRANSACTIONS string